realm/realm-library/src/objectServer/java/io/realm/internal/network/ExponentialBackoffTask.java

```diff
@@ -69,7 +78,7 @@ public void run() {
                 onSuccess(response);
                 break;
             } else {
-                if (shouldAbortTask(response)) {
+                if (shouldAbortTask(response) || attempt == maxRetries + 1) {
```

Comment:
what about if the network drops and reconnect? is it handled by the other logic?

Reply:
yes